File: packages/schemas/ts/content.ts
import { z } from 'zod';

export const subjectSummarySchema = z.object({
  subject: z.string(),
  section_count: z.number().int().nonnegative(),
  sample_grade: z.string().nullable(),
});
export type SubjectSummary = z.infer<typeof subjectSummarySchema>;

export const subjectListSchema = z.array(subjectSummarySchema);

export const contentSectionSchema = z.object({
  subject: z.string(),
  grade: z.string().nullable(),
  title: z.string(),
  body_md: z.string(),
  page_start: z.number().int().nullable(),
  page_end: z.number().int().nullable(),
  chunk_index: z.number().int().nullable().optional(),
});
export type ContentSection = z.infer<typeof contentSectionSchema>;

export const contentSectionsPageSchema = z.object({
  items: z.array(contentSectionSchema),
  total: z.number().int().nonnegative(),
  page: z.number().int().positive(),
  page_size: z.number().int().positive(),
});
export type ContentSectionsPage = z.infer<typeof contentSectionsPageSchema>;

export const bagrutExamSchema = z.object({
  display_name: z.string(),
  file_url: z.string(),
  year: z.number().int().nullable(),
  exam_type: z.string(),
});
export type BagrutExam = z.infer<typeof bagrutExamSchema>;

export const bagrutExamListSchema = z.array(bagrutExamSchema);

File: packages/schemas/schemas/content.py
"""Public free-tier content browsing schemas."""

from __future__ import annotations

from pydantic import BaseModel, Field


class SubjectSummary(BaseModel):
    subject: str
    section_count: int = 0
    sample_grade: str | None = None


class ContentSection(BaseModel):
    subject: str
    grade: str | None = None
    title: str
    body_md: str
    page_start: int | None = None
    page_end: int | None = None
    chunk_index: int | None = None


class ContentSectionsPage(BaseModel):
    items: list[ContentSection]
    total: int
    page: int
    page_size: int


class BagrutExam(BaseModel):
    display_name: str
    file_url: str
    year: int | None = None
    exam_type: str


class BagrutExamList(BaseModel):
    items: list[BagrutExam]
